google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/StorageSettings.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.storage
+
+final class StorageSettings private (val projectId: String, val clientEmail: String, val privateKey: String) {
+  def withProjectId(projectId: String): StorageSettings = copy(projectId = projectId)
+
+  def withClientEmail(clientEmail: String): StorageSettings = copy(clientEmail = clientEmail)
+
+  def withPrivateKey(privateKey: String): StorageSettings = copy(privateKey = privateKey)
+
+  private def copy(projectId: String = projectId,
+                   clientEmail: String = clientEmail,
+                   privateKey: String = privateKey): StorageSettings =
+    new StorageSettings(projectId, clientEmail, privateKey)
+
+  override def toString: String =
+    s"StorageSettings(projectId=$projectId, clientEmail=$clientEmail, privateKey=$privateKey"
```

Comment:
Don't print out the key by mistake...
```suggestion
     s"StorageSettings(projectId=$projectId, clientEmail=$clientEmail, privateKey=***)"
```